From 68a9b448f89f00cc5c37873d11a000d7442c9f82 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 08 Jun 2022 11:18:34 +0300
Subject: [PATCH] Revert "[SW-66675] gaudi3: add tpm erom routine"

This reverts commit 4000fd083dd67d7957dabf266f990dc3ee9bc1fd.

Change-Id: Ia776414628a4275506c953c98c70116b29e0cf80
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 358137c..9a3b548 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -705,14 +705,7 @@
 		return;
 
 	if (!security_efuse_tpm_loc()) {
-		if (!boot_desc_init(HL_FW_BOOT_PTR_EROM)) {
-			arcboot_rom_status_update(
-					PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
-			return;
-		}
-
-		/* Load TPM binary from EROM */
-		ext_rom_load_erom_image(HL_FW_EROM_TPM);
+		/* TODO: Add erom tpm routine*/
 	} else {
 		arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_TPM);
 
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index 1a79908..abb2a27 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -28,6 +28,5 @@
 uint32_t arcboot_rom_get_arc_addr(uint64_t addr);
 void arcboot_rom_wait_for_fuse_data(void);
 int arcboot_rom_fuse_single_die_config(void);
-void ext_rom_load_erom_image(enum hl_fw_img_type_erom img_type);
 void disable_spi_gw(void);
 uint64_t get_pcie_pll_freq(enum pll_div_out div);
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 230a34b..cd06706 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -25,7 +25,7 @@
 	return DMA_DEST_HBW;
 }
 
-void ext_rom_load_erom_image(uint32_t index)
+static void ext_rom_load_erom_image(uint32_t index)
 {
 	enum arcboot_rom_progress img_sts_base = PROGRESS_ARCBOOT_EROM_IMG0;
 	enum dma_error dma_err = DMA_SUCCESS;
